operator-common/src/main/java/io/strimzi/operator/common/Annotations.java

```diff
@@ -67,10 +69,13 @@
     );
 
     private static Map<String, String> annotations(ObjectMeta metadata) {
-        Map<String, String> annotations = metadata.getAnnotations();
-        if (annotations == null) {
-            annotations = new HashMap<>(3);
-            metadata.setAnnotations(annotations);
+        Map<String, String> annotations = new HashMap<>();
+        if (metadata != null) {
```

Comment:
What are the situations when metadata is null? I would not expect this to happen since every resource needs to have a name.

Reply:
Mock tests.

Reply:
That sounds like mock tests are crap if the resources don't even have a name. Shouldn't the mock tests be fixed first?